api/src/main/java/org/infinispan/api/sync/Counter.java

```diff
@@ -0,0 +1,58 @@
+package org.infinispan.api.sync;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 13.0
+ **/
+public interface Counter {
+
+   long value();
+
+   /**
+    * Increments the counter.
+    */
+   default void increment() {
+      add(1L);
+   }
+
+
+   /**
+    * Decrements the counter.
+    */
+   default void decrement() {
+      add(-1L);
+   }
+
+   /**
+    * Adds the given value to the new value.
+    *
+    * @param delta the value to add.
+    */
+   void add(long delta);
+
+   /**
+    * Resets the counter to its initial value.
+    */
+   void reset();
+
+   /**
+    * Returns the name of this counter
+    *
+    * @return the name of this counter
+    */
+   String name();
+
+   /**
+    * Returns an async version of this counter
+    *
+    * @return
+    */
+   org.infinispan.api.async.Counter async();
+
+   /**
+    * Returns a reactive version of this counter
+    *
+    * @return
+    */
+   org.infinispan.api.reactive.Counter reactive();
```

Comment:
maybe remove these methods? users can use the Counters entry point to retrieve whatever interface they want.